### PR TITLE
Reader: Properly handle feeds that come back with a new feed URL

### LINKED ISSUE #13934: Reader: Following an external RSS site results in two entries in Manage
Start at https://wpcalypso.wordpress.com/following/manage

Best to start with a fairly empty account.

Follow `http://daringfireball.net`.

Notice the entry appear in the list.

Wait.

Reload.

Notice we now have two entries.

This is happening because the feed URL changes in the response...

### PR DESCRIPTION
**Pull request by blowery**
When following some sites using direct URL entry, the returned feed URL will not match what the user entered. This can happen for a variety of reasons, including auto-discovery of feeds and protocol normalization.

Fixes #13934

### CODE REVIEW

**Comment by blowery** [comment]
Redoing this to move the logic to the reducer

**Comment by blowery** [comment]
and, redone

File: client/state/reader/follows/reducer.js [hunk 1]
```diff
@@ -96,23 +96,32 @@ export const items = createReducer(
...
-			// Reset follow error state
-			if ( state[ urlKey ] && state[ urlKey ].error ) {
+			const actualFeedUrl = get(
+				action.payload,
+				[ 'follow', 'feed_URL' ],
+				action.payload.feedUrl
+			);
+
+			const newState = { ...state };
+			if ( actualFeedUrl !== action.payload.feedUrl ) {
```
Comment (samouri) [inline client/state/reader/follows/reducer.js]: add a single comment here saying:
`the case where a user follows by url and the actual feed_URL is different from what the user entered`

**Comment by samouri** [comment]
Code LGTM!

File: client/state/reader/follows/reducer.js (hunk 1 above)
Comment (blowery) [inline client/state/reader/follows/reducer.js]: you have commit rights... ;)

**Comment by bluefuton** [comment]
Shall we run wp-prettier on our bit of the state tree? Quite a lot of unrelated formatting changes in this PR.

**Comment by bluefuton** [comment]
@blowery when I do follow `http://daringfireball.com`, it does appear in my subs eventually but with the wrong following state, weirdly:

screen shot 2017-05-12 at 16 14 58

**Comment by samouri** [comment]
did some rebasing/cherry picking here.  i'm commandeering this PR with @blowery's permissions :)

File: client/state/reader/follows/reducer.js (hunk 1 above)
Comment (samouri) [inline client/state/reader/follows/reducer.js]: yes, yes i do.

**Comment by samouri** [comment]
> @blowery when I do follow http://daringfireball.com, it does appear in my subs eventually but with the wrong following state, weirdly:

i'm not seeing that.  what I am seeing though is that the "follow daringfireball.com" loses its green after we move it in the reducer.

on top of that, the placeholder doesn't get updated until rv gets prodded a bit.  gonna need to fix that as well.

**Comment by samouri** [comment]
Solved the first issue. We now remember things on a best-effort bases, but until we do the full fetch when a user types in the .com address, we can't have a complete solution there.

Now working on the placeholder-staying-too-long issue.

**Comment by samouri** [comment]
I think I've solved all the issues! Please give it another set of 👁 @bluefuton / @blowery .

To Test:
 - [ ] test following by typing in a url. make sure that you can follow, it appears, the placeholder acts properly, and you can follow/unfollow from either the subs item or the search results. 
- [ ] I changed the way react-virtualized realizes when to refresh.  so make sure that searching things, adding things, changing sort, etc still work as expected

**Comment by blowery** [comment]
LGTM. I can't approve it though.

**Pull request merged**